### ISSUE TITLE
missing IIIF links for some BNF-EN newspaper issue

### ISSUE THREAD

**Issue by mromanello** [impresso/impresso-text-acquisition#102]
(issue reported by @ehoelzl)

- the IIIF link for some issues cannot be found via the BNF API (following [this procedure](http://api.bnf.fr/documents-de-presse-numerises-en-mode-article-du-projet-europeana-newspapers#chapitre5), see also note below)
- this issue affects only 34 (out of ~107K) newspaper issues (full list here below)

**Note from @ehoelzl on retrieval of IIIF links**:

> Basically there is a URL per journal (e.g. https://gallica.bnf.fr/services/Issues?ark={ark}/date) where the ark differs per journal. This URL gives us all the years for this journal. Then we can query for each year (https://gallica.bnf.fr/services/Issues?ark={ark}/date&date={year})  and get the ark link for each issue. Examples: https://gallica.bnf.fr/services/Issues?ark=cb39294634r/date and https://gallica.bnf.fr/services/Issues?ark=cb39294634r/date&date=1932

----
```
jdpl-1932-12-30-b
jdpl-1932-12-30-c
jdpl-1933-07-01-b
legaulois-1899-08-07-b
legaulois-1899-08-14-b
legaulois-1899-08-16-b
legaulois-1899-08-17-b
legaulois-1899-08-18-b
legaulois-1899-08-19-b
legaulois-1899-08-21-b
legaulois-1899-08-25-b
legaulois-1899-08-26-b
legaulois-1899-08-28-b
legaulois-1899-08-29-b
legaulois-1899-08-30-b
legaulois-1899-08-31-b
legaulois-1899-09-04-b
legaulois-1899-09-05-b
legaulois-1899-09-07-b
legaulois-1899-09-09-b
lepetitparisien-1896-03-12-a
oecaen-1914-08-26-a
oecaen-1914-08-27-a
oecaen-1914-08-28-a
oecaen-1914-08-29-a
oecaen-1914-08-30-a
oecaen-1914-08-31-a
oecaen-1939-08-29-a
oecaen-1940-04-24-a
oecaen-1940-04-25-a
oerennes-1901-12-19-a
oerennes-1901-12-30-a
lepji-1896-05-24-a
lepji-1909-12-05-a
```

**Comment by mromanello** [comment]
This issue may have occurred due to a performance issue on BNF's IIIF servers, thus when running again the importer it may disappear.

**Comment by piconti** [comment]
Update on the current status of availability of the issues on the Gallica iiif API.

It seems that while the issues and corresponding files exist in our data, they are simply missing from the Gallica online collection (cannot be found on the API or website). 
At the individual issue level, may be due to an error internal to Gallica or BNF, but it's not possible to access the iiif link or arkid.
- eg. `lepetitparisien-1896-03-12-a` –> [see API output](https://gallica.bnf.fr/services/Issues?ark=cb34419111x/date&date=1896), also [see webapp](https://gallica.bnf.fr/ark:/12148/cb34419111x/date1896), here march 13 is duplicated, but not was to say it's linked to the missing 12th.
- eg. `oecaen-1914-08-26` to `oecaen-1914-08-31-a` –> [see API output](https://gallica.bnf.fr/services/Issues?ark=cb41193642z/date&date=1914) and [see webapp](https://gallica.bnf.fr/ark:/12148/cb41193642z/date1914).

This seems to be the case regardless of the edition, as different editions are missing, and examples of accessible second or third editions exist ([see weebapp](https://gallica.bnf.fr/ark:/12148/cb34419111x/date19231108), [API output](https://gallica.bnf.fr/services/Issues?ark=cb34419111x/date&date=1923) and corresponding images [1](https://gallica.bnf.fr/iiif/ark:/12148/bpt6k605561k/f1/full/full/0/native.jpg), [2](https://gallica.bnf.fr/iiif/ark:/12148/bpt6k605562z/f1/full/full/0/native.jpg), with it's not the case for `jdpl-1932-12-30` here, where editions b and c are not available (see : https://gallica.bnf.fr/ark:/12148/cb39294634r/date19331230).

**In conclusion**: _Re-ingesting the data will not solve this issue while it's not fixed on Gallica's side_. Since the issues are still not available after three years, it's probable that their unavailability is not due to a simple technical issue, unless it was unnoticed. 
It would make sense to communicate this list of issues to them and keep track of the evolution of the list.

**Comment by piconti** [comment]
I have not found any email or exchange with Jean-Philippe from BNF about this, @e-maud can you confirm it's the same on your side?

Do we want to ignore these few titles or contact them to try to see if it's simply an API bug on their side?
